vmm/src/memory_manager.rs

```diff
@@ -485,37 +495,55 @@ impl MemoryManager {
         let end_of_device_area = GuestAddress(mmio_address_space_size() - 1);
 
         let mut start_of_device_area = MemoryManager::start_addr(guest_memory.last_addr(), false);
+        let mut virtiomem_regions: Vec<Arc<GuestRegionMmap>> = Vec::new();
 
-        let mut virtiomem_region = None;
-        let mut virtiomem_resize = None;
-        if let Some(size) = config.hotplug_size {
-            if config.hotplug_method == HotplugMethod::VirtioMem {
-                // Alignment must be "natural" i.e. same as size of block
-                let start_addr = GuestAddress(
-                    (start_of_device_area.0 + virtio_devices::VIRTIO_MEM_DEFAULT_BLOCK_SIZE - 1)
-                        / virtio_devices::VIRTIO_MEM_DEFAULT_BLOCK_SIZE
-                        * virtio_devices::VIRTIO_MEM_DEFAULT_BLOCK_SIZE,
-                );
+        if !use_zones {
```

Comment:
So `use_zones` is really `user_provided_zones` since you create a zone and add it to `zones` even when no user provided zone is given...

Reply:
Yes exactly. Let me change the variable name to clarify this.